src/openrct2/world/TileInspector.cpp

```diff
@@ -512,11 +512,11 @@ int32_t tile_inspector_surface_toggle_corner(int32_t x, int32_t y, int32_t corne
 
     if (flags & GAME_COMMAND_FLAG_APPLY)
     {
-        const uint8_t originalSlope = surfaceElement->properties.surface.slope;
+        const uint8_t originalSlope = surfaceElement->AsSurface()->GetSlope();
         const bool diagonal = (originalSlope & TILE_ELEMENT_SLOPE_DOUBLE_HEIGHT) >> 4;
 
-        surfaceElement->properties.surface.slope ^= 1 << cornerIndex;
-        if (surfaceElement->properties.surface.slope & TILE_ELEMENT_SLOPE_ALL_CORNERS_UP)
+        surfaceElement->AsSurface()->SetSlope(surfaceElement->AsSurface()->GetSlope() ^ (1 << cornerIndex));
```

Comment:
I'd prefer non-trivial arguments to be computed in a temporary variable on their own line.